shared/js/background/chrome-events.es6.js

```diff
@@ -603,6 +709,62 @@ chrome.webRequest.onBeforeSendHeaders.addListener(
     extraInfoSpecSendHeaders
 )
 
+/**
+ * Click to Load
+ */
+
+/*
+ * On FireFox, redirecting to a JS surrogate in some cases causes a CORS error. Determine if that is the case here.
+ * If so, and we have an alternate XRAY surrogate implementation, inject it.
+ */
+chrome.webRequest.onBeforeRedirect.addListener(
+    details => {
+        const tab = tabManager.get({ tabId: details.tabId })
+        if (tab && !tab.site.isBroken && !tab.site.whitelisted && details.responseHeaders && trackerutils.facebookExperimentIsActive()) {
+            // Detect cors error
+            const headers = details.responseHeaders
+            const corsHeaders = [
+                'Access-Control-Allow-Origin'
+            ]
+            const corsFound = headers.filter(v => corsHeaders.includes(v.name)).length
+
+            if (corsFound && details.redirectUrl) {
+                const xray = trackerutils.getXraySurrogate(details.redirectUrl)
+                if (xray && utils.getBrowserName() === 'moz') {
+                    console.log('Normal surrogate load failed, loading XRAY version')
+                    chrome.tabs.executeScript(details.tabId, {
+                        file: `public/js/content-scripts/${xray}`,
+                        matchAboutBlank: true,
+                        frameId: details.frameId,
+                        runAt: 'document_start'
+                    })
+                }
+            }
+        }
+    },
+    { urls: ['<all_urls>'] },
+    ['responseHeaders']
+)
+
+// Inject our content script to overwite FB elements
+chrome.webNavigation.onCommitted.addListener(details => {
+    const tab = tabManager.get({ tabId: details.tabId })
+    if (tab && tab.site.isBroken) {
+        console.log('temporarily skip embedded object replacements for site: ' + details.url +
+          'more info: https://github.com/duckduckgo/content-blocking-whitelist')
```

Comment:
We changed this repo.
```suggestion
          'more info: https://github.com/duckduckgo/content-blocking-lists')
```